runners/spark/src/test/java/org/apache/beam/runners/spark/SparkNativePipelineVisitorTest.java

```diff
@@ -0,0 +1,234 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.spark;
+
+import static org.junit.Assert.assertThat;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import org.apache.beam.runners.spark.examples.WordCount;
+import org.apache.beam.runners.spark.translation.EvaluationContext;
+import org.apache.beam.runners.spark.translation.SparkPipelineTranslator;
+import org.apache.beam.runners.spark.translation.TransformTranslator;
+import org.apache.beam.runners.spark.translation.streaming.StreamingTransformTranslator;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.coders.KvCoder;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.TextIO;
+import org.apache.beam.sdk.io.kafka.KafkaIO;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.Distinct;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.Flatten;
+import org.apache.beam.sdk.transforms.GroupByKey;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.transforms.Sum;
+import org.apache.beam.sdk.transforms.WithKeys;
+import org.apache.beam.sdk.transforms.windowing.FixedWindows;
+import org.apache.beam.sdk.transforms.windowing.Window;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionList;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.streaming.api.java.JavaStreamingContext;
+import org.hamcrest.Matchers;
+import org.joda.time.Duration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+
+/**
+ * Test {@link SparkNativePipelineVisitor} with different pipelines.
+ */
+public class SparkNativePipelineVisitorTest {
+
+  private File outputDir;
+
+  private JavaSparkContext jsc;
+
+  @Rule
+  public final transient PipelineRule batchPipelineRule = PipelineRule.batch();
+
+  @Rule
+  public final transient PipelineRule streamingPipelineRule = PipelineRule.streaming();
+
+  @Rule
+  public final TemporaryFolder tmpDir = new TemporaryFolder();
+
+  @SuppressWarnings("ResultOfMethodCallIgnored")
+  @Before
+  public void setUp() throws IOException {
+    outputDir = tmpDir.newFolder("out");
+    outputDir.delete();
+    jsc = new JavaSparkContext("local[*]", "Existing_Context");
```

Comment:
In tests, default is to reuse context, and there's a rule to force Reuse (yes/no) as well so I don't think you need to specifically relate to the context in `before`/`teardown`